redislock.go

```diff
@@ -95,7 +105,20 @@ func (c *Client) Obtain(ctx context.Context, key string, ttl time.Duration, opt
 }
 
 func (c *Client) obtain(ctx context.Context, key, value string, ttl time.Duration) (bool, error) {
-	return c.client.SetNX(ctx, key, value, ttl).Result()
+	ok, err := c.client.SetNX(ctx, key, value, ttl).Result()
+
+	if !ok {
+		storedkey, err := c.client.Get(ctx, key).Result()
```

Comment:
This means that you can call Obtain twice and it will grant you access to the lock both times.

Reply:
I don't quite understand why you want to use redislock for this scenario. You already have a message queue to ensure distributed locking, right? 

1. a process shifts a message message `M1` containing `{ id: "O3Fi", action: "create project", name: "duck" }`. It calls `SETNX duck O3Fi` re-emits `M1`
2. process shifts `M1` and checks status, project creation is still pending, it re-emits the message
3. process shifts `M2` containing `{ id: "s00Z", action: "update project", name: "duck", what: "ever" }`, it tries `SETNX duck s00Z` which fails, so it re-emits `M2` to the queue.
4. process shifts `M1` and checks status, project creation is finished, it calls `DEL duck`
5. process shifts `M2` ...

I don't see why you would need a distributed, time-constrained lock here.

Reply:
In my scenario, the messages are handled by completely different processes.

In step 4 the process cannot delete the previous lock (so it cannot call DEL duck), because it cannot get the previous lock object. Just like in the Issue here: https://github.com/bsm/redislock/issues/43

Reply:
My big issue with this PR is the `SetNX` then `Get` bit. It introduces a race-condition and makes the library prone to user mistakes (and a tad slower for everyone, even if they are not using a custom token). I don't think it makes sense here. 

I still don't believe you need an atomic, distributed lock for use case you describe. You can achieve this without atomicity as long as you don't have two processes working on the same message at the same time (which should never happen if you use a proper message queue)